transferui/src/main/java/com/hyperwallet/android/ui/transfer/viewmodel/CreateTransferViewModel.java

```diff
@@ -0,0 +1,338 @@
+ * including without limitation the rights to use, copy, modify, merge, publish, distribute,
+ * sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT
+ * NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package com.hyperwallet.android.ui.transfer.viewmodel;
+
+import static com.hyperwallet.android.model.transfermethod.HyperwalletTransferMethod.TransferMethodFields.TOKEN;
+import static com.hyperwallet.android.model.transfermethod.HyperwalletTransferMethod.TransferMethodFields.TRANSFER_METHOD_CURRENCY;
+
+import android.text.TextUtils;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.MutableLiveData;
+import androidx.lifecycle.Observer;
+import androidx.lifecycle.ViewModel;
+import androidx.lifecycle.ViewModelProvider;
+
+import com.hyperwallet.android.model.HyperwalletErrors;
+import com.hyperwallet.android.model.transfer.Transfer;
+import com.hyperwallet.android.model.transfermethod.HyperwalletTransferMethod;
+import com.hyperwallet.android.model.user.HyperwalletUser;
+import com.hyperwallet.android.ui.common.repository.Event;
+import com.hyperwallet.android.ui.transfer.repository.TransferRepository;
+import com.hyperwallet.android.ui.transfermethod.repository.TransferMethodRepository;
+import com.hyperwallet.android.ui.user.repository.UserRepository;
+
+import java.util.UUID;
+
+/**
+ * Create Transfer View Model
+ */
+public class CreateTransferViewModel extends ViewModel {
+
+    private static final String CLIENT_IDENTIFICATION_PREFIX = "HW-MBLA-";
+
+    private final TransferRepository mTransferRepository;
+    private final TransferMethodRepository mTransferMethodRepository;
+    private final UserRepository mUserRepository;
+
+    private final MutableLiveData<HyperwalletTransferMethod> mTransferDestination = new MutableLiveData<>();
+    private final MutableLiveData<Boolean> mTransferAvailableFunds = new MutableLiveData<>();
+    private final MutableLiveData<Boolean> mIsLoading = new MutableLiveData<>();
+    private final MutableLiveData<Boolean> mIsCreateQuoteLoading = new MutableLiveData<>();
+    private final MutableLiveData<Transfer> mQuoteAvailableFunds = new MutableLiveData<>();
+    private final MutableLiveData<Transfer> mQuoteTransfer = new MutableLiveData<>();
+    private final MutableLiveData<String> mTransferAmount = new MutableLiveData<>();
```

Comment:
Since we are using LiveData to manage data lifecycle, might as well use this in data form fields, which in this turn give us the ability to delegate android lifecycle destruction/construction of data to LiveData framework (we don't need to manage phone rotation, app back stack and app sleep mode), this also in turn gives us the advantage of easy maintainability since there's only 1 single truth of data source which is our ViewModel (No need to look over the lifecycle listener where the data changed, we always know that its from view model + component)

Reply:
👍